pkg/proxy/proxy.go

```diff
@@ -87,11 +94,12 @@ func getProxy(namespace string, name string, serviceCAfile string ) *httputil.Re
 		reverseProxy.FlushInterval = time.Millisecond * 100
 		reverseProxy.Transport = transport
 		reverseProxy.ModifyResponse = FilterHeaders
+		cacheManager.SetProxy(proxiesMapKey, reverseProxy)
```

Comment:
It seems we are filling the cache but is not being invalidated. Maybe in a following PR we could add a TTL